students/scott/session03/mailroom.py

```diff
@@ -1,75 +1,87 @@
-Jim = [4567, 1999, 43213]
-Martha = [3500000, 49]
-
-
-# Prompt the user for what they want to do
-
-# Define some variables
-
-choice = 0
-name = ""
-
-## Text menu in Python
-
-def print_menu():  # These are the options in the menu
-    print(20* "*" +' MENU ' + 20* "*")
-    print("1. Send a thank you note")
-    print("2. Create a report")
-    print("3. Quit program")
-    print(20* "*" +' MENU ' + 20* "*")
-
-def get_action(): # this is getting the input from the user on what they want to do
-    print_menu()
-    choice = input("What would you like to do? ")
-
-def take_action():    
-    choice = int(input("What would you like to do? "))
-    if choice == 1:
-        print("Let's send a thank you note \n")
-        name = input("Who do you want to send a thank you note to? ")
-        get_name()
-    elif choice == 2:
-        print("Let's create a report")
-    elif choice == 3:
-        print("Ending program")
-        exit()
-    else:
-        print("That isn't a valid selection. Please try again")
-
-
-# Get the name from the user; if the name isn't in the donors list, add it
-def get_name():
-    name = input("Who do you want to send a thank you note to? Enter a Full Name, or 'list' to get a list of donors. ")
-    if name in donors:
-        print(name)
-    elif name == "list":
-        print(donors)
-        name = input("Which of those would you like to send a thank you note to? ")
-    else:
-        donors.append(name)
-
-# Write a thank you note to a specific donor
+donors_list = [ ("Paul", [100000, 1200000, 750000, 25000]), 
+                ("Jerry", [20000, 43126]), 
+                ("Clark", [1292, 6788, 128, 9827]), 
+                ("Terry", [11, 2000000, 1000001, 45]), 
+                ("Arthur", [50, 100] ), 
+                ("Jim", [4567, 1999, 43213]), 
+                ("Martha", [3500000, 49])]
+
+
+def interactive_loop():
+
+    if __name__ == "__main__":
+        program_status = True
+        while program_status:
+            selection = start_menu()
+            if selection == '1':
+                get_donor()
+            elif selection == '2':
+                report()
+            elif selection == '3':
+                program_status = False
+            else:
+                print("Invalid entry. Please select 1, 2 or 3")
+
+
+def start_menu():  # These are the options in the menu
+    selection = input('''
+        Make a selection--
+
+        '1. Send a thank you note'
+        '2. Create a report'
+        '3. Quit program'
+
+        ''')
+    return selection
+
+
+#print the names from the donors list, without the amounts of their donations
+
+def print_list():
+    for donor in donors_list:
+        print(donor[0])
+
+
+# Get the name from the user
+
+# I tried using a while loop here, but can't get my head around how to order everything.....
+# for some reason, if 'menu' is entered, and the user is kicked back to the start menu, entering '3' DOESN'T exit the program and I have no idea why
```

Comment:
see the start_menu function above -- all it does is ask for the user's selection -- it  doesn't do anything with it.

Reply:
I think you want to call "interactive_loop()" here instead.

alternatively, have this function return when you leave the loop, and it should drop back to interactive_loop()